contracts/Controller.sol

```diff
@@ -398,21 +398,21 @@ contract Controller is Initializable, OwnableUpgradeSafe, ReentrancyGuardUpgrade
 
     /**
      * @dev return if an expired oToken contract’s settlement price has been finalized
-     * @param _otoken address of the oToken
+     * @param _underlying oToken underlying asset
+     * @param _strike oToken strike asset
+     * @param _collateral oToken collateral asset
+     * @param _expiry otoken expiry timestamp
      * @return True if the oToken has expired AND all oracle prices at the expiry timestamp have been finalized, False if not
      */
-    function isSettlementAllowed(address _otoken) public view returns (bool) {
-        OtokenInterface otoken = OtokenInterface(_otoken);
-
-        address underlying = otoken.underlyingAsset();
-        address strike = otoken.strikeAsset();
-        address collateral = otoken.collateralAsset();
-
-        uint256 expiry = otoken.expiryTimestamp();
-
-        bool isUnderlyingFinalized = oracle.isDisputePeriodOver(underlying, expiry);
-        bool isStrikeFinalized = oracle.isDisputePeriodOver(strike, expiry);
-        bool isCollateralFinalized = oracle.isDisputePeriodOver(collateral, expiry);
+    function isSettlementAllowed(
+        address _underlying,
+        address _strike,
+        address _collateral,
+        uint256 _expiry
+    ) public view returns (bool) {
```

Comment:
hmm this may not be the most user-friendly. Should we user test this with devs first?

Reply:
Why it is not ?